From d8e68e67b397aa247109bae639ed72fecf480fa3 Mon Sep 17 00:00:00 2001
From: Matthijs van Duin <matthijsvanduin@gmail.com>
Date: Thu, 19 Jul 2018 07:52:12 +0200
Subject: [PATCH] pty: fix O_CLOEXEC for TIOCGPTPEER

It was being ignored because the flags weren't passed to fd allocation.

Fixes: 54ebbfb16034 ("tty: add TIOCGPTPEER ioctl")
Signed-off-by: Matthijs van Duin <matthijsvanduin@gmail.com>
---
 drivers/tty/pty.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/tty/pty.c b/drivers/tty/pty.c
index 64338442050e..8782b3dea49a 100644
--- a/drivers/tty/pty.c
+++ b/drivers/tty/pty.c
@@ -622,7 +622,7 @@ int ptm_open_peer(struct file *master, struct tty_struct *tty, int flags)
 	if (tty->driver != ptm_driver)
 		return -EIO;
 
-	fd = get_unused_fd_flags(0);
+	fd = get_unused_fd_flags(flags);
 	if (fd < 0) {
 		retval = fd;
 		goto err;
-- 
2.18.0

